packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1441,10 +1798,47 @@ export class TransactionController extends BaseControllerV1<
    * Ensure `releaseLock` is called once processing of the `nonce` value is complete.
    *
    * @param address - The hex string address for the transaction.
+   * @param networkClientId - The network client ID for the transaction, used to fetch the correct nonce tracker.
    * @returns object with the `nextNonce` `nonceDetails`, and the releaseLock.
    */
-  async getNonceLock(address: string): Promise<NonceLock> {
-    return this.nonceTracker.getNonceLock(address);
+  async getNonceLock(
+    address: string,
+    networkClientId?: NetworkClientId,
+  ): Promise<NonceLock> {
+    // TODO(JL): SmartTransactionController reaches into TransactionController.nonceTracker directly. Should probably change this.
+    // TODO(JL): Revisit this method. It's a bit complicated and not obvious what it achieves.
```

Comment:
note these elsewhere and then delete the comments

Reply:
working on this

Reply:
[``](https://github.com/MetaMask/MetaMask-planning/issues/1019)

Reply:
https://github.com/MetaMask/core/pull/3643/commits/ea2fbe1266bbeb01f6299fa8273f44f870c51c89